hal/nrf_nfct.h

```diff
@@ -267,6 +267,40 @@ typedef enum
     NRF_NFCT_SELRES_PROTOCOL_NFCDEP_T4AT = 3,  /**< NFC-DEP Protocol and Type 4A Tag platform). */
 } nrf_nfct_selres_protocol_t;
 
+/**
+ * @brief RFU7 (bit b8) configuration for the SEL_RES frame according to
+ *        the NFC Forum Digital Protocol Technical Specification.
+ */
+typedef enum
+{
+    NRF_NFCT_SELRES_RFU7_0 = 0,
```

Comment:
Shouldn't we define those bits value like 0,1,2,3 in the MDK @nika-nordic ?
I don't sure about this because those bits are RFU.

Reply:
I tried to follow how it's done for eg. `nrf_nfct_selres_protocol_t`, but maybe that was not correct since this is RFU?

Reply:
After discussion with @KAGA164 I suggest to rework API for setting RFU bits. IMHO current approach introduces too much code. What about introducing simple, single, new API:
```
/**
 * @brief Function for setting the SEL_RES frame configuration.
 *
 * @details The SEL_RES frame is handled automatically by the NFCT hardware.
 *
 * @param[in] p_reg  Pointer to the structure of registers of the peripheral.
 * @param[in] selres SEL_RES frame configuration.
 */
void nrf_nfct_selres_set(NRF_NFCT_Type * p_reg,
                         uint32_t        selres);

(...)

void nrf_nfct_selres_set(NRF_NFCT_Type * p_reg,
                         uint32_t        selres);
{
    p_reg->SELRES = selres;
}
```

And analogous getter API.

What do you think @petterssonandreas ?

Reply:
I agree, that the current approach introduces a lot of code, where most of it is just boilerplate. I could definitely be OK with your proposal to consolidate into a single simple function, however I do see some problems with it:

- Currently one can use named enums, eg. the members of `nrf_nfct_selres_protocol_t` to set the protocol. How should we handle that going forward? Removing all named enums will make other code less understandable without additions of defines or comments. Or should we keep the named enums (modify them to be at the correct bit position) and allow users to OR them together?
- Currently, the cascade bit is not exposed in a setter, only in a getter (but it's called `check` in the code). The cascade bit is not intended to be set by any user, but is automatically set by the hardware. But maybe the same argument as the RFU bits apply to that, that why not allow setting it, maybe it could be useful in some protocol (although I doubt that would ever be the case).
- Should the mdk be updated? Currently it has masks and positions for all of the different fields.

Reply:
What I would like to really avoid is to explicitly use term `RFU` in the API as theoretically those bits shouldn't be set by the user according to spec.

Idea behind `nrf_nfct_selres_set(p_reg, selres)` is that user can specify anything in `selres` (so it covers your use-case) without direct exposure of the option of RFU bits setting.

Answering your questions:
- we can introduce new enum type `nrf_nfct_selres_protocol_mask_t` that contains protocol masks which can be ORed with custom bitmask (possibly defined in your application_ containing RFU setting and then provided to the `nrf_nfct_selres_set` API. I would like to avoid introducing named enums for RFU bits as this will directly imply that they are configurable.
- not sure if I understand correctly, but you would like to have the setter for cascade bit? Documentation says it is controlled by hardware, so direct setter shouldn't be available.
- I am almost sure MDK won't be updated with this as it is tightly coupled with the Product Specification, and according to Product Specification these bits should not be modified.

Sorry if this looks confusing, but please remember that we need to follow Product Specification when defining the nrf_* APIs. IMHO approach with `nrf_nct_selres_set` is a fine compromise

Reply:
Sure, I get your point of not naming something RFU, to not indicate that those can be set. However, this is assuming we *need* to follow the NFC Forum Specification, one could argue that Nordic would want to encourage the users of its products to even be able to develop their own NFC protocols. But I guess that's another discussion, and we should not go into that now, as that would require a much bigger overhaul of the HAL.

- OK, that sounds fine.
- No, I meant that currently there is no setter for the cascade bit, and that is very intentional I think. But with the `nrf_nct_selres_set` API it would provide a way to actually set the cascade bit. Is that bad? Dangerous? Or acceptable since it won't be advertised?
- OK, but the MDK does mention RFU bit positions and masks. So it's not avoided there. So I was just thinking if we wanted some coherency with this change. (But I haven't really used the MDK, so not sure how that is determined or used, so I guess we can skip that for now, but I wanted to make you aware of that.)

Sure, I just wanted to clarify some things, and sadly I don't really know which Product Specification you are referring to, so can't really follow that. So it's great that you tell me how you want things done :slightly_smiling_face: 

Also, another thing, changing the API would be a non-backwards compatible change. Is that acceptable?

Reply:
Product Specification ( `PS` ) I am referring to is a document describing the SoC, available on the Nordic Infocenter (example [link](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf52840%2Fkeyfeatures_html5.html) for the nRF52840)

- glad you agree :)
- as for the cascade bit I think we can mask the provided `selres` parameter by ORed masks from the MDK:
```
void nrf_nfct_selres_set(p_reg, uint32_t selres)
{
    p_reg->SELRES = selres & (NFCT_SELRES_RFU7_Msk | NFCT_SELRES_RFU43_Msk | NFCT_SELRES_RFU10_Msk |
                              NFCT_SELRES_PROTOCOL_Msk );
)
```
- I will ask internally for such change in the MDK, but even if it is accepted it would take a long time until it will be updated. So for this particular PR we need to assume such symbols won't be available.

We can't introduce any API breaking changes, so I suggest to introduce `nrf_nfct_selres_set/get()` as a new API and leave all existing APIs untouched

Reply:
Oh, thanks for the link.

OK, good, I think we have then agreed on what to do. I will do a fix and hopefully that will correlate to what you meant.

Reply:
Did a fix!